lib/l10n.js

```diff
@@ -1,29 +1,74 @@
+/**
+ * Compose user messages.
+ */
 
-'use strict';
+// Internal packages:
+const rulesWrapper = require('./rules-wrapper')
+,   english = require('./l10n-en_GB')
+,   spanish = require('./l10n-es_ES')
+;
 
-var messages = {
-    es_ES: require("./l10n-es_ES").messages
-,   en_GB: require("./l10n-en_GB").messages
-,   selectors: require("./l10n-selectors").selectors
-,   wording: require("./rules")
+// Constants:
+const rules = rulesWrapper.rules
+,   en_GB = english.messages
+,   es_ES = spanish.messages
+;
+
+// Variables:
+var lang;
+
+/**
+ * Set a locale to be used globally by this module.
+ *
+ * @param {String} language - locale, expressed as a string, eg <code>en_GB</code>.
+ */
+
+exports.setLanguage = function(language) {
+    if (!language)
+        throw new Error('l10n.setLanguage() invoked without passing a language code as parameter');
+    else if (language.match(/^en([\.\-_]?gb)?$/i))
+        lang = en_GB;
+    else if (language.match(/^es([\.\-_]?es)?$/i))
+        lang = es_ES;
+    else
+        throw new Error('Language code passed to l10n.setLanguage() is not valid');
 };
 
-exports.message = function (lang, rule, key, extra) {
-    if (!messages[lang]) return "@@@No such language: " + lang + "@@@";
-    var l10n = messages[lang][rule + "." + key];
-    if (!l10n) return "@@@No such entry: " + rule + "." + key + "@@@";
-    if (extra) {
-        return l10n.replace(/\$\{([^\}]+)\}/g, function (m, p1) {
-            if (extra.hasOwnProperty(p1)) return extra[p1];
-            return "@@@No such data: ${" + p1 + "}@@@";
-        });
-    }
-    if (messages.selectors && messages.wording) {
-        var selector = messages.selectors[rule + "." + key];
-        if (selector) {
-            l10n += ' <br> ' + eval('messages.wording.' + selector);
+/**
+ * @TODO Document.
+ */
+
+exports.message = function (profile, rule, key, extra) {
+    const result = {};
+    var name;
+    if ('string' === typeof rule)
+        name = rule;
+    else
+        name = rule.name;
+    if (!lang)
+        throw new Error('l10n.message() invoked before a locale is defined; call l10n.setLanguage() first');
+    else if (!rulesWrapper.hasOwnProperty(profile))
+        throw new Error(`l10n.message(): unknown profile code “${profile}”`);
```

Comment:
Checking a document with the wrong profile makes the app crash. For instance, a WD with the profile Rec gives:
```
Error: l10n.message(): unknown profile code “Rec”
    at Object.exports.message (.../specberus/lib/l10n.js:51:15)
    at null.<anonymous> (.../specberus/app.js:96:37)
    at emitTwo (events.js:105:20)
    at emit (events.js:185:7)
    at Specberus.error (.../specberus/lib/validator.js:138:23)
    at err (.../specberus/lib/rules/headers/dl.js:28:16)
    at Object.exports.check (.../specberus/lib/rules/headers/dl.js:94:14)
    at .../specberus/lib/validator.js:107:18
    at Array.forEach (native)
    at doValidation (.../specberus/lib/validator.js:103:23)
```

These errors should be caught so the app doesn't crash.